src/state/ducks/Quotes.ts

```diff
@@ -0,0 +1,224 @@
+import Immutable from 'immutable'
+import { createSelector } from 'reselect'
+import * as sdk from '@kiltprotocol/sdk-js'
+import errorService from '../../services/ErrorService'
+import KiltAction from '../../types/Action'
+import { State as ReduxState } from '../PersistentStore'
+import * as Wallet from './Wallet'
+import { IMyIdentity } from '../../types/Contact'
+
+export type IQuoteEntry = sdk.IQuoteAttesterSigned | sdk.IQuoteAgreement // Could find a better name for this
+
+function hash(quote: sdk.IQuote): string {
+  const quoteHash = {
+    attesterAddress: quote.attesterAddress,
+    cTypeHash: quote.cTypeHash,
+    cost: quote.cost,
+    currency: quote.currency,
+    timeframe: quote.timeframe,
+    termsAndConditions: quote.termsAndConditions,
+  }
+  return sdk.Crypto.hashObjectAsStr(JSON.stringify(quoteHash))
+}
+
+interface ISaveAction extends KiltAction {
+  payload: {
+    quoteId: Entry['quoteId']
+    claimerAddress: string
+    quote: IQuoteEntry
+  }
+}
+
+interface ISaveAgreedQuoteAction extends KiltAction {
+  payload: {
+    quoteId: Entry['quoteId']
+    quote: IQuoteEntry
+  }
+}
```

Comment:
I currently can't see in the quotes view that a quote was agreed to - do we set this?

Reply:
Yes, we set it in the `SubmitTerms.tsx` - `SaveQuote`. This is the bug I fixed that would save the Quote when pressing create new Quote.

Reply:
sorry, should have made this clearer - there are two different actions, one called `saveQuote`, one called `saveAgreedQuote`, but in the quotes list I can't tell which is a `Quote` and which an `AgreedQuote` - I wonder whether that should not be indicated with a little symbol or something

Reply:
I understand now. I have updated it so, now it reflects what quote it is saving.

`saveAttestersQuote`